testing/test_junitxml.py

```diff
@@ -1,20 +1,52 @@
 import os
 import platform
 from datetime import datetime
+from pathlib import Path
 from xml.dom import minidom
 
 import py
+import xmlschema
 
 import pytest
 from _pytest.junitxml import LogXML
 from _pytest.reports import BaseReport
 
 
-def runandparse(testdir, *args):
-    resultpath = testdir.tmpdir.join("junit.xml")
-    result = testdir.runpytest("--junitxml=%s" % resultpath, *args)
-    xmldoc = minidom.parse(str(resultpath))
-    return result, DomNode(xmldoc)
+@pytest.fixture(scope="session")
+def schema():
+    """Returns a xmlschema.XMLSchema object for the junit-10.xsd file"""
+    fn = Path(__file__).parent / "example_scripts/junit-10.xsd"
+    with fn.open() as f:
+        return xmlschema.XMLSchema(f)
+
+
+@pytest.fixture(name="run_and_parse")
+def run_and_parse_(testdir, schema):
+    """
+    Fixture that returns a function that can be used to execute pytest and return
+    the parsed ``DomNode`` of the root xml node. It accepts a ``validate`` argument,
+    in which case it will configure pytest to write with the xunit2 family and validate
+    against the schema.
+    """
+
+    def run(*args, validate=False):
+        if validate and not testdir.tmpdir.join("tox.ini").isfile():
+            testdir.makeini(
+                """
+                [pytest]
+                junit_family=xunit2
+            """
+            )
+
+        resultpath = testdir.tmpdir.join("junit.xml")
+        result = testdir.runpytest("--junitxml=%s" % resultpath, *args)
+        if validate:
+            with resultpath.open() as f:
+                schema.validate(f)
+        xmldoc = minidom.parse(str(resultpath))
+        return result, DomNode(xmldoc)
+
+    return run
```

Comment:
What's the benefit of having parametrizing with validate=True/False, if it could just validate it then?  (I see no benefit of having the test without validation, if it gets done anyway in another test)
I.e. I suggest having separate fixtures `run_and_parse` and `run_validate_and_parse` (where the latter could use the former).
This would allow to just use one of them, and remove the `@pytest.mark.parametrize("validate", [True, False])` instances.

Reply:
> What's the benefit of having parametrizing with validate=True/False, if it could just validate it then? (I see no benefit of having the test without validation, if it gets done anyway in another test)

We only validate xunit2, we don't have validation for xunit1. You will notice that when `validate=True`, I "force" to use xunit2 family.

But thinking about this a bit more, I think it is more explicit to have `run_and_parse(family)` with `family` being "xunit1" or "xunit2", and validation being implicitly done for `xunit2`. What do you thik?

Reply:
I went ahead and made the above changes, it is now much clearer IMHO (thanks @blueyed).

@asottile would you like to review this as well?

Reply:
Great!